### PR TITLE
 add getIsSaved function to get isSaved field

### PR DESCRIPTION
**Pull request by hewonjeong**
-  add getIsSaved function to get isSaved field

### CODE REVIEW

File: src/server/items/models.js [hunk 1]
```diff
@@ -43,5 +43,37 @@ export default class ItemManager {
...
+        }
+        for (let j = pointer; j < items.length; j = j + 1) {
+          items[j].isSaved = false;
+        }
+        resolve(items);
+      });
+    });
+  }
+  static getFields(array, field) {
+    const ans = array.map((obj)=>{
```
Comment (ovekyc) [inline src/server/items/models.js]: The variable ans seems like not necessary.
You can just return array.map()

File: src/server/items/models.js [hunk 2]
```diff
@@ -43,5 +43,33 @@ export default class ItemManager {
...
+      return new Promise((resolve)=>{
+        posts.sort((a, b) => {
+          return a.key > b.key;
+        });
+        const itemKeys = this.getFields(items, 'key');
+        const postKeys = this.getFields(posts, 'key');
+        let pointer = 0;
+        for (let i = 0; i < itemKeys.length; i = i + 1) {
+          const idx = postKeys.indexOf(itemKeys[i], pointer);
+          if (idx !== -1) {
```
Comment (ochanje210) [inline src/server/items/models.js:62]: what does -1 means?

File: src/server/items/models.js [hunk 3]
```diff
@@ -43,5 +43,33 @@ export default class ItemManager {
...
+    .then((posts)=>{
+      return new Promise((resolve)=>{
+        posts.sort((a, b) => {
+          return a.key > b.key;
+        });
+        const itemKeys = this.getFields(items, 'key');
+        const postKeys = this.getFields(posts, 'key');
+        let pointer = 0;
+        for (let i = 0; i < itemKeys.length; i = i + 1) {
+          const idx = postKeys.indexOf(itemKeys[i], pointer);
```
Comment (hewonjeong) [inline src/server/items/models.js:61]: @TODO
we can improve its performance more using binaryIndexOf or saving most close index here.

File: src/server/items/models.js (hunk 2 above)
Comment (hewonjeong) [reply src/server/items/models.js:62]: It means there is no itemKeys[i] value in postKeys[pointerits∽length].
if it has value, idx is index of value in postKeys.

File: src/server/items/models.js (hunk 3 above)
Comment (ochanje210) [reply src/server/items/models.js:61]: why don't you put a comment of this todo in the code?

File: src/server/items/models.js (hunk 2 above)
Comment (ochanje210) [reply src/server/items/models.js:62]: thanks

**Comment by festinalente91** [comment]
lgtm

**Comment by ovekyc** [comment]
lgtm

**Pull request merged**